R/assertions.R

```diff
@@ -132,30 +133,41 @@ assert_data_frame <- function(arg,
 assert_character_scalar <- function(arg,
                                     values = NULL,
                                     case_sensitive = TRUE,
-                                    optional = FALSE) {
+                                    optional = FALSE,
+                                    arg_name = rlang::caller_arg(arg),
+                                    message = NULL,
```

Comment:
I think we need to rethink the `message` argument. It works well for assertions which issue a single message only. But if different messages are issued like "the argument is of wrong type" and "the argument is of correct type but the values are not as expected", it does not work.

Maybe we should remove the `message` and `class` argument. And if it it necessary to change the message or class, the error should be caught and modified.

Reply:
After meeting to discuss, we will keep the `message` argument. Thanks for the good discussion point!